docs/workflow.md

```diff
@@ -0,0 +1,137 @@
+                       --dns-zone ${dnsZone} \
+                       --email ${email} \
+                       --authz-domain ${authzDomain} \
+                       --project ${gceProject} \
+                       --oauth-client-id ${oauthClientId} \
+                       --oauth-client-secret ${oauthClientSecret}
+```
+
+The `--only-generate` command-line flag tells `kubeprod` to configure the underlying platform (in this example, GKE) by creating the DNS zone if necessary, service accounts, etc. Afterwards, the root manifest and JSON configuration files are generated and `kubeprod` exits cleanly. At this point the operator can inspect the changes introduced by `kubeprod` and perform any customisations as explained next.
+
+### Implement customizations (optional)
+
+The root manifest file, `kubeprod-manifest.jsonnet`, generated by `kubeprod` is the place where to implement any customisations desired for BKPR. For example, as described in [cert-manager](components.md#cert-manager) it is possible to use a different Let's Encrypt environment, in particular when testing BKPR.
+
+For example, the `kubeprod-manifest.jsonnet` as created by `kubeprod` looks like this:
+
+```
+# Cluster-specific configuration
+(import "manifests/platforms/gke.jsonnet") {
+    config:: import "kubeprod-autogen.json",
+    // Place your overrides here
+}
+```
+
+If we evaluate the manifest using `kubecfg` and filter the output searching for the Let's Encrypt environment, we will find this:
+
+```
+$ kubecfg show kubeprod-manifest.jsonnet | grep -- --default-issuer-name
+        - --default-issuer-name=letsencrypt-prod
+```
+
+To customise BKPR and use the Let's Encrypt staging environment, then `kubeprod-manifest.jsonnet` has to be modified like this:
+
+```
+# Cluster-specific configuration
+(import "manifests/platforms/gke.jsonnet") {
+    config:: import "kubeprod-autogen.json",
+    // Place your overrides here
+    cert_manager+: {
+        letsencrypt_environment:: "staging",
+    }
+}
+```
+
+What the previous block of code does is the following:
+
+1. Imports the manifest at `manifests/platforms/gke.jsonnet` which describes how to deploy BKPR on GKE.
+1. Under the `cert_manager` key (which implements the `cert-manager` component), overrides the `letsencrypt_environment` property to `staging` from its default value of `prod`.
+
+To ensure the override is working as expected let's evaluate the manifest again using `kubecfg` and filter the output searching for the Let's Encrypt environment. This time we will see that `cert-manager` will use the staging environment instead of the production environment:
+
+```
+$ kubecfg show kubeprod-manifest.jsonnet | grep -- --default-issuer-name
+        - --default-issuer-name=letsencrypt-staging
+```
+
+Please refer to the [components](componens.md) documentation for supoorted customisations in every component and also feel free to read the BKPR source code. After all, that's one of the advatages of being open source.
```

Comment:
I would remove:  and also feel free to read the BKPR source code. After all, that's one of the advatages of being open source.

I would also add a link to jsonnet.org